src/Parsing/ParserState.php

```diff
@@ -220,18 +220,26 @@ public function parseCharacter($bIsForIdentifier)
     }
 
     /**
-     * @return array<int, Comment>|void
+     * Consumes whitespace and comments and returns the comments.
+     * If $withoutComment is true, only whitespace is consumed.
+     *
+     * @param bool $withoutComment Do not consume comments, only whitespace.
+     *
+     * @return array<int, Comment>|void List of comments.
```

Comment:
We can be even more specific (and omit the void return type, which is incorrect for this method and also is not allowed in union types):

```suggestion
     * @return list<Comment> list of comments
```